modules/programs/alot.nix

```diff
@@ -0,0 +1,145 @@
+
+  alot_accounts = config.accounts.email.accounts;
+
+  # https://alot.readthedocs.io/en/latest/configuration/contacts_completion.html
+  contactCompletionStr = let
+      contactCompletion = "notmuch address";
+    in
+    account:
+    ''
+      [[[abook]]]
+      type = shellcommand
+      command = '' + (
+    if contactCompletion == "notmuch address" then ''
+      '${pkgs.bash}/bin/bash -c "${pkgs.notmuch}/bin/notmuch  address --format=json --output=recipients  date:1Y.."'
+    regexp = '\[?{"name": "(?P<name>.*)", "address": "(?P<email>.+)", "name-addr": ".*"}[,\]]?'
+    shellcommand_external_filtering = False
+    '' else if contactCompletion == "notmuch --config ${notmuchConfig account} address simple" then
+    "'${pkgs.notmuch}/bin/notmuch address --format=json date:1Y..'"
+    else "");
+
+  sendMailCommand = account:
+  if account.alot ? sendMailCommand then account.alot.sendMailCommand
+  else if account.msmtp.enable then programs.msmtp.sendCommand account else "";
+
+  accountStr = {name, userName, address, realName, ...} @ account:
+    let
+      showSignature = "none";
+    in
+    ''
+      [[${name}]]
+      address=${address}
+      realname=${realName}
+      sendmail_command = ${sendMailCommand account}
+      signature_as_attachment = ${if showSignature == "attach" then "True" else "False"}
+
+      # contact completion
+      ${contactCompletionStr account}
+      ${if showSignature == "attach" then "gpg_key = {account." else ""}
+    ''
+    ;
+
+configFile = mailAccounts:
+    pkgs.writeText "alot.conf" (''
+# Generated by home-manager
+# see http://alot.readthedocs.io/en/latest/configuration/config_options.html
+hooksfile = "$XDG_CONFIG_HOME/alot/hm_hooks"
+${cfg.extraConfig}
+
+[bindings]
+${cfg.bindings}
+
+[accounts]
+
+${concatStringsSep "\n" (map accountStr mailAccounts)}
+'');
+
+in
+{
+
+  options = {
+    programs.alot = {
+      enable = mkEnableOption "Alot";
+
+      extraHookFile = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Content to append to the hooks file (after home-manager python code)
+        '';
+      };
+
+      contactCompletionCommand = mkOption {
+        default = contactCompletionStr;
+        description = "Can override what is decided in contactCompletion";
+      };
+
+      bindings = mkOption {
+        type = types.lines;
+        default = ''
+          % = refresh;
+        '';
+        description = "Can override what is decided in contactCompletion";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = ''
+          auto_remove_unread = True;
+          ask_subject = False;
+          handle_mouse = "True;
+          initial_command = "search tag:inbox AND NOT tag:killed";
+          input_timeout = 0.3;
+          prefer_plaintext = True;
+          thread_indent_replies = 4;
+        '';
+        description = ''
+          Extra configuration lines to add to ~/.config/alot/config.
```

Comment:
Either need some markup here, e.g.,

    Extra configuration lines to add to <filename>~/.config/alot/config</filename>.

or rewording it, e.g.,

    Extra lines to add to the alot configuration file.